clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -149,6 +152,11 @@ struct LoweringPreparePass : public LoweringPrepareBase<LoweringPreparePass> {
   SmallVector<mlir::Attribute, 4> globalCtorList;
   /// List of dtors to be called when unloading module.
   SmallVector<mlir::Attribute, 4> globalDtorList;
+  /// List of annotations in the module, need to be a vector of Ops
+  /// as we need view to annotated global values to create global annotation var
+  /// later in LLVM lowering.
+  //  One global value could have multiple AnnotateAttr
+  std::vector<std::pair<mlir::Attribute, mlir::Operation *>> globalAnnotations;
```

Comment:
This should be:
```
SmallVector<mlir::Attribute, 8> globalAnnotations;
```
Each Attribute here should be an ArrayAttr with the form: `[StringAttr, AnnotateAttr]`. If you do this, then you just need to pass globalAnnotations directly when creating the attribute to attach to `cir.global_annotations`.

Note that instead of keeping track of `mlir::Operation *` you should retrieve the string directly at the point you walk the global/functions.